indexing-service/src/main/java/org/apache/druid/indexing/seekablestream/RecordSupplierInputSource.java

```diff
@@ -68,29 +69,35 @@ public RecordSupplierInputSource(
     this.recordSupplier = recordSupplier;
     this.useEarliestOffset = useEarliestOffset;
     this.iteratorTimeoutMs = iteratorTimeoutMs;
-    try {
-      assignAndSeek(recordSupplier);
-    }
-    catch (InterruptedException e) {
-      throw new SamplerException(e, "Exception while seeking to partitions");
-    }
+
+    assignAndSeek(recordSupplier);
   }
 
   private void assignAndSeek(RecordSupplier<PartitionIdType, SequenceOffsetType, RecordType> recordSupplier)
-      throws InterruptedException
   {
-    final Set<StreamPartition<PartitionIdType>> partitions = recordSupplier
-        .getPartitionIds(topic)
-        .stream()
-        .map(partitionId -> StreamPartition.of(topic, partitionId))
-        .collect(Collectors.toSet());
-
-    recordSupplier.assign(partitions);
-
-    if (useEarliestOffset) {
-      recordSupplier.seekToEarliest(partitions);
-    } else {
-      recordSupplier.seekToLatest(partitions);
+    try {
+      final Set<StreamPartition<PartitionIdType>> partitions = recordSupplier
+          .getPartitionIds(topic)
+          .stream()
+          .map(partitionId -> StreamPartition.of(topic, partitionId))
+          .collect(Collectors.toSet());
+
+      recordSupplier.assign(partitions);
+
+      if (useEarliestOffset) {
+        recordSupplier.seekToEarliest(partitions);
+      } else {
+        recordSupplier.seekToLatest(partitions);
+      }
+    }
+    catch (Exception e) {
+      throw new SamplerException(
+          e,
+          "Exception while seeking to the [%s] offset of partitions in topic [%s]: %s",
+          useEarliestOffset ? "earliest" : "latest",
```

Comment:
Neat!

Reply:
Just noticed something: this class is shared by both kaka and kinesis streaming. In Kinesis vernacular, it's sequence number instead of offset. The Druid supervisor settings[ are also different](https://druid.apache.org/docs/latest/development/extensions-core/kinesis-ingestion.html ) for Kafka vs Kinesis: `useEarliestSequenceNumber` vs `useEarliestOffset`.

If it's easy to identify that here, we could perhaps change the messaging a tiny bit. What do you think, @amaechler?